CompilerSource/parser/precedence.h

```diff
@@ -0,0 +1,118 @@
+      {TT_SCOPEACCESS,  {Precedence::kScope,          Associativity::LTR}},
+      {TT_DOT,          {Precedence::kMemberAccess,   Associativity::LTR}},
+      {TT_ARROW,        {Precedence::kMemberAccess,   Associativity::LTR}},
+      {TT_DOT_STAR,     {Precedence::kMemberPointer,  Associativity::LTR}},
+      {TT_ARROW_STAR,   {Precedence::kMemberPointer,  Associativity::LTR}},
+      {TT_STAR,         {Precedence::kMultiply,       Associativity::LTR}},
+      {TT_SLASH,        {Precedence::kDivide,         Associativity::LTR}},
+      {TT_DIV,          {Precedence::kDivide,         Associativity::LTR}},
+      {TT_MOD,          {Precedence::kModulo,         Associativity::LTR}},
+      {TT_PERCENT,      {Precedence::kModulo,         Associativity::LTR}},
+      {TT_PLUS,         {Precedence::kAdd,            Associativity::LTR}},
+      {TT_MINUS,        {Precedence::kSubtract,       Associativity::LTR}},
+      {TT_LSH,          {Precedence::kShift,          Associativity::LTR}},
+      {TT_RSH,          {Precedence::kShift,          Associativity::LTR}},
+      {TT_THREEWAY,     {Precedence::kThreeWayComp,   Associativity::LTR}},
+      {TT_LESS,         {Precedence::kRelational,     Associativity::LTR}},
+      {TT_LESSEQUAL,    {Precedence::kRelational,     Associativity::LTR}},
+      {TT_GREATER,      {Precedence::kRelational,     Associativity::LTR}},
+      {TT_GREATEREQUAL, {Precedence::kRelational,     Associativity::LTR}},
+      {TT_EQUALTO,      {Precedence::kEquality,       Associativity::LTR}},
+      {TT_NOTEQUAL,     {Precedence::kEquality,       Associativity::LTR}},
+      {TT_AMPERSAND,    {Precedence::kBitAnd,         Associativity::LTR}},
+      {TT_CARET,        {Precedence::kBitXOr,         Associativity::LTR}},
+      {TT_PIPE,         {Precedence::kBitOr,          Associativity::LTR}},
+      {TT_AND,          {Precedence::kBoolAnd,        Associativity::LTR}},
+      {TT_XOR,          {Precedence::kBoolXOr,        Associativity::LTR}},
+      {TT_OR,           {Precedence::kBoolOr,         Associativity::LTR}},
+      {TT_ASSIGN,       {Precedence::kAssign,         Associativity::RTL}},
+      {TT_EQUALS,       {Precedence::kAssign,         Associativity::RTL}},
+      {TT_ASSOP,        {Precedence::kCompoundAssign, Associativity::RTL}},
+      {TT_COMMA,        {Precedence::kComma,          Associativity::LTR}},
+
+      // TODO: TT_THROW as kThrow RTL
+      // TODO: TT_ kYield RTL
+  };
+
+  std::unordered_map<TokenType, OperatorPrecedence> Precedence::kUnaryPostfixPrec {
+      {TT_INCREMENT,    {Precedence::kUnaryPostfix, Associativity::LTR}},
+      {TT_DECREMENT,    {Precedence::kUnaryPostfix, Associativity::LTR}},
+  };
+
+  std::unordered_map<TokenType, OperatorPrecedence> Precedence::kTernaryPrec {
+      {TT_QMARK, {Precedence::kTernary, Associativity::RTL}},
+  };
+
+  std::unordered_map<TokenType, OperatorPrecedence> Precedence::kSpecialPrec {
```

Comment:
This is meant for operators that don't really fit into any of the other categories, like array subscript being technically binary but with an extra token at the end, and function call being an n-ary operator where the opening parentheses is an "infix" operator.